cfme/tests/automate/test_service_dialog.py

```diff
@@ -63,95 +77,103 @@ def test_service_dialog_duplicate_name(appliance, request):
+            'ele_desc': fauxfactory.gen_alphanumeric(),
+            'choose_type': "Tag Control"
+        },
+        'options': {
+            'field_category': "Service Level",
+            'field_required': "Yes"
+        }
     }
     dialog = create_dialog(appliance, element_data)
     request.addfinalizer(dialog.delete_if_exists)
 
 
 def test_textareabox_dialog_element(appliance, request):
+    if appliance.version >= "5.9":
+        choose_type = "Text Area"
+    else:
+        choose_type = "Text Area Box",
     element_data = {
-        'ele_label': "ele_" + fauxfactory.gen_alphanumeric(),
-        'ele_name': fauxfactory.gen_alphanumeric(),
-        'ele_desc': fauxfactory.gen_alphanumeric(),
-        'choose_type': "Text Area Box",
-        'field_required': True
+        'element_information': {
+            'ele_label': "ele_" + fauxfactory.gen_alphanumeric(),
+            'ele_name': fauxfactory.gen_alphanumeric(),
+            'ele_desc': fauxfactory.gen_alphanumeric(),
+            'choose_type': choose_type,
+        },
+        'options': {
+            'field_required': "Yes"
+        }
     }
     dialog = create_dialog(appliance, element_data)
     request.addfinalizer(dialog.delete_if_exists)
 
 
 def test_reorder_elements(appliance, request):
     element_1_data = {
-        'ele_label': "ele_" + fauxfactory.gen_alphanumeric(),
-        'ele_name': fauxfactory.gen_alphanumeric(),
-        'ele_desc': fauxfactory.gen_alphanumeric(),
-        'choose_type': "Text Box",
-        'default_text_box': "Default text"
+        'element_information': {
+            'ele_label': "ele_label_" + fauxfactory.gen_alphanumeric(),
+            'ele_name': fauxfactory.gen_alphanumeric(),
+            'ele_desc': fauxfactory.gen_alphanumeric(),
+            'choose_type': "Text Box",
+        }
     }
     element_2_data = {
-        'ele_label': "ele_" + fauxfactory.gen_alphanumeric(),
-        'ele_name': fauxfactory.gen_alphanumeric(),
-        'ele_desc': fauxfactory.gen_alphanumeric(),
-        'choose_type': "Check Box",
-        'default_value': True,
-        'field_required': True,
+        'element_information': {
+            'ele_label': "ele_" + fauxfactory.gen_alphanumeric(),
+            'ele_name': fauxfactory.gen_alphanumeric(),
+            'ele_desc': fauxfactory.gen_alphanumeric(),
+            'choose_type': "Check Box",
+
+        }
     }
     service_dialog = DialogCollection(appliance)
-    sd = service_dialog.create(label='label_' + fauxfactory.gen_alphanumeric(),
-        description="my dialog", submit=True, cancel=True,)
+    if appliance.version >= "5.9":
+        sd = service_dialog.create(label='label_' + fauxfactory.gen_alphanumeric(),
+                                   description="my dialog")
+    else:
+        sd = service_dialog.create(label='label_' + fauxfactory.gen_alphanumeric(),
+                                   description="my dialog", submit=True, cancel=True)
```

Comment:
This may be a better thing to address with a followup PR, but there are a lot of instances of a version check to instantiate a service_dialog, adding submit/cancel kwargs for older versions.

Have you looked at the potential for setting a default for these optional arguments when instantiating against 58z? May allow for removing version checks, at least when using the defaults for submit/cancel.

Reply:
Kinda spitballing - but another option is to always pass submit/cancel and drop the args (with logged warning) in the instantiate method after checking for >5.9

Reply:
Done